### PR TITLE
Mon 16533 add performance indicator

### PR DESCRIPTION
**Pull request by jean-christophe81**
## Description

add bench test and a python program bench_plot.py to visualize bench results

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [X] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: tests/broker-engine/muxer_filter.robot [hunk 1]
```diff
@@ -298,20 +303,25 @@ CBD_RELOAD_AND_FILTERS
...
     Should Be True    ${result}    msg=No event rejected by the rrd output whereas only storage category is enabled.
 
     Log To Console    Third configuration: all events are sent.
     # New configuration
     Broker Config Output Remove    central    centreon-broker-master-rrd    filters
+    ${start}=    Get Current Date
     Restart Engine
     Reload Broker
-    ${start}=    Get Current Date
+    #wait broker reload
```
Comment (thearchit3ct) [inline tests/broker-engine/muxer_filter.robot:320]:
```suggestion
    # wait broker reload
```

File: tests/broker-engine/bench.robot [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+
+
+*** Test Cases ***
+BENCH_1000STATUS
+    [Documentation]    external command CHECK_SERVICE_RESULT 1000 times
+    [Tags]    broker    engine    bench
+    Config Engine    ${1}    ${50}    ${20}
+    # We want all the services to be passive to avoid parasite checks during our test.
+    Set Services passive    ${0}    service_.*
```
Comment (thearchit3ct) [inline tests/broker-engine/bench.robot:27]: To standardize the title, keyword Should Be Proper/Title Case
```suggestion
    Set Services Passive    ${0}    service_.*
```

File: tests/broker-engine/bench.robot [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    Broker Config Add Item    central    bbdo_version    3.0.0
+    Broker Config Add Item    rrd    bbdo_version    3.0.0
+    Config Broker Sql Output    central    unified_sql
+    ${start}=    Get Current Date
+    Start Broker  
+    Start Engine
+    ${content}=    Create List    check_for_external_commands
+    ${result}=    Find In Log with Timeout    ${engineLog0}    ${start}    ${content}    60
+    Should Be True    ${result}    msg=No check for external commands executed for 1mn.
+    ${broker_stat_before}=  get_broker_process_stat  51001
```
Comment (thearchit3ct) [inline tests/broker-engine/bench.robot:44]:
```suggestion
    ${broker_stat_before}  Get Broker Process Stat  51001
```

File: tests/broker-engine/delete-poller.robot [hunk 4]
```diff
@@ -348,8 +348,14 @@ EBDP5
     ${result}=    Find In Log with Timeout    ${engineLog0}    ${start}    ${content}    60
     Should Be True    ${result}    msg=check_for_external_commands is missing.
 
+    ${remove_time}=  Get Current Date
```
Comment (thearchit3ct) [inline tests/broker-engine/delete-poller.robot:351]:
```suggestion
    ${remove_time}  Get Current Date
```

File: tests/README.md [hunk 5]
```diff
@@ -50,6 +50,10 @@ And it is also possible to execute a specific test, for example:
 robot broker/sql.robot
 ```
 
+In order to execute bench tests (broker-engine/bench.robot), you need also to install py-cpuinfo, cython, unqlite and boto3
+
+pip3 install py-cpuinfo cython unqlite gitpython boto3
+
```
[the remark this answers is not in the document]
Comment (jean-christophe81) [reply tests/README.md:56]: resolved

**Pull request merged**